### PR TITLE
Add IdGenerator interface

### PR DESCRIPTION
**Pull request by lalex**
Add SDK IdGenerator interface and default RandomIdGenerator implementation.

Spec: https://github.com/open-telemetry/opentelemetry-specification/blob/03a6f0fd9d1d2f5e08858b102ba620ebf3e4a6fa/specification/trace/sdk.md#id-generators

### CODE REVIEW

File: sdk/Trace/IdGenerator.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+interface IdGenerator
+{
+    public function generateTraceId(): string;
```
Comment (bobstrecansky) [inline sdk/Trace/IdGenerator.php:9]: In the spec, these are labeled `generateSpanIdBytes` and `generateTraceIdBytes`  - should we follow suit with the spec here?

File: sdk/Trace/RandomIdGenerator.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        try {
+            return bin2hex(random_bytes(intdiv($hexLength, 2)));
+        } catch (\Throwable $e) {
+            return $this->fallbackAlgorithm($hexLength);
+        }
+    }
+
+    private function fallbackAlgorithm(int $hexLength): string
+    {
+        return substr(str_shuffle(str_repeat('0123456789abcdef', $hexLength)), 1, $hexLength);
```
Comment (bobstrecansky) [inline sdk/Trace/RandomIdGenerator.php:33]: I think this is fine for now, but we may want to consider another option here in the future.  I think it'd be a very strange use case where `bin2hex` doesn't return a correct value.

**Review by bobstrecansky — COMMENTED** [review]
A couple small questions and a couple psalm errors, but I think this PR is off to a great start!

File: sdk/Trace/IdGenerator.php (hunk 1 above)
Comment (lalex) [reply sdk/Trace/IdGenerator.php:9]: In the spec they show an example from Java implementation, where they also have a dedicated classes `TraceId` and `SpanId` for this. I don't think we have `bytes` here in PHP yet, for now Trace ID and Span ID are plain strings. In such case more suitable name is `generateTraceIdString` which sounds too obvious.

File: sdk/Trace/RandomIdGenerator.php (hunk 2 above)
Comment (lalex) [reply sdk/Trace/RandomIdGenerator.php:33]: The fallback method guards a rare case when `random_bytes()` may throw an error (https://www.php.net/manual/en/function.random-bytes.php).
The implementations of this method was copied from the [existing code](https://github.com/open-telemetry/opentelemetry-php/blob/96d0feb2d23dc9c42aef9587f4260a9e174a05bb/sdk/Trace/SpanContext.php#L186).

**Comment by lalex** [comment]
It seems the psalm errors are not related to the changes from the PR. I think it's better to fix it in separate PR.

File: sdk/Trace/RandomIdGenerator.php (hunk 2 above)
Comment (bobstrecansky) [reply sdk/Trace/RandomIdGenerator.php:33]: Good enough for me!

**Comment by bobstrecansky** [comment]
The psalm errors seem to be related to https://github.com/vimeo/psalm/issues/5203

I'll add a PR to address this.

**Review by bobstrecansky — APPROVED** [review]

**Pull request merged**